lib/ansible/modules/cloud/amazon/cloudtrail.py

```diff
@@ -603,9 +592,9 @@ def main():
         if module.check_mode:
             acct_id = '123456789012'
             try:
-                sts_client = boto3_conn(module, conn_type='client', resource='sts', region=region, endpoint=ec2_url, **aws_connect_params)
+                sts_client = module.client('sts')
                 acct_id = sts_client.get_caller_identity()['Account']
-            except ClientError:
+            except (BotoCoreError, ClientError):
                 pass
```

Comment:
Is this pass statement right now? I wonder if the original author of these couple lines of code meant to pass on a certain error code (which would occur with ClientError). BotoCoreErrors should usually just be fatal. I didn't dig into git history though.

Reply:
This fragment is only fetching the current AccountID while running in check_mode and is only used to generate a 'fake' ARN.  My thoughts were that since we don't explode in most cases, we might as well do our best to continue.

Reply:
Sounds reasonable! Thanks for looking into it. I'll look back through this again over lunch in a few hours and merge. Thanks for all your work cleaning these things up.